CMakeLists.txt

```diff
@@ -27,64 +27,203 @@
-    ./include/dropt.h
-    ./include/dropt_string.h
+add_library(${PROJECT_NAME}
+    ${${PROJECT_NAME}_c_files}
+    ${${PROJECT_NAME}_h_files}
 )
 
-set(dropt_c_files
-    ./src/dropt.c
-    ./src/dropt_handlers.c
-    ./src/dropt_string.c
+target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include> $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+
+cpack_add_component(c_lib
+	DISPLAY_NAME "${PROJECT_NAME}"
+	DESCRIPTION "The C library"
+)
+cpack_add_component(c_lib_dev
+	DISPLAY_NAME "${PROJECT_NAME} SDK"
+	DESCRIPTION "SDK for the C library"
+)
+set(CPACK_DEBIAN_C_LIB_PACKAGE_NAME "lib${PROJECT_NAME}")
+
+set(CPACK_DEBIAN_C_LIB_DEV_PACKAGE_NAME "${CPACK_DEBIAN_C_LIB_PACKAGE_NAME}-dev")
+set(CPACK_DEBIAN_C_LIB_DEV_PACKAGE_SUGGESTS "cmake, pkg-config, pkg-conf")
+set(CPACK_DEBIAN_C_LIB_DEV_PACKAGE_DEPENDS "${CPACK_DEBIAN_C_LIB_PACKAGE_NAME}")
+
+set(CPACK_RPM_C_LIB_PACKAGE_NAME "${CPACK_DEBIAN_C_LIB_PACKAGE_NAME}")
+
+set(CPACK_RPM_C_LIB_DEV_PACKAGE_NAME "${CPACK_RPM_C_LIB_PACKAGE_NAME}-devel")
+set(CPACK_RPM_C_LIB_DEV_PACKAGE_SUGGESTS "${CPACK_DEBIAN_PACKAGE_SUGGESTS}")
+set(CPACK_RPM_C_LIB_DEV_PACKAGE_DEPENDS "${CPACK_RPM_C_LIB_PACKAGE_NAME}")
+
+install(TARGETS ${PROJECT_NAME}
+	EXPORT ${PROJECT_NAME}
+	LIBRARY
+		DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+		COMPONENT c_lib
+	ARCHIVE
+		DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+		COMPONENT c_lib_dev
+	INCLUDES
+		DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+		# COMPONENT c_lib_dev # component is not allowed for includes! Headers are installed separately! Includes only marks the headers for export
+)
+install(FILES ${${PROJECT_NAME}_h_files}
+	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+	COMPONENT c_lib_dev
 )
 
-add_library(dropt
-    ${dropt_c_files}
-    ${dropt_h_files}
+install(EXPORT ${PROJECT_NAME}
+	FILE "${PROJECT_NAME}Config.cmake"
+	NAMESPACE dropt::
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
+	COMPONENT c_lib_dev
 )
+configure_pkgconfig_file(${PROJECT_NAME} c_lib_dev)
 
-target_include_directories(dropt PUBLIC ${CMAKE_CURRENT_LIST_DIR}/include)
+set(${PROJECT_NAME}xx_hpp_files
+    ${IncludeDir}/${PROJECT_NAME}xx.hpp
+)
 
-set(droptxx_hpp_files
-    ./include/droptxx.hpp
+set(${PROJECT_NAME}xx_cpp_files
+    ${SrcDir}/${PROJECT_NAME}xx.cpp
 )
 
-set(droptxx_cpp_files
-    ./src/droptxx.cpp
+cpack_add_component(cpp_lib
+	DISPLAY_NAME "${PROJECT_NAME}xx"
+	DESCRIPTION "The C++ library"
 )
+cpack_add_component(cpp_lib_dev
+	DISPLAY_NAME "${PROJECT_NAME}xx SDK"
+	DESCRIPTION "SDK for the C++ library"
+)
+set(CPACK_DEBIAN_CPP_LIB_PACKAGE_NAME "lib${PROJECT_NAME}xx")
+set(CPACK_DEBIAN_CPP_LIB_PACKAGE_DEPENDS "${CPACK_DEBIAN_C_LIB_PACKAGE_NAME}")
+
+set(CPACK_DEBIAN_CPP_LIB_DEV_PACKAGE_NAME "${CPACK_DEBIAN_CPP_LIB_PACKAGE_NAME}-dev")
+set(CPACK_DEBIAN_CPP_LIB_DEV_PACKAGE_DEPENDS "${CPACK_DEBIAN_CPP_LIB_PACKAGE_NAME}, ${CPACK_DEBIAN_C_LIB_DEV_PACKAGE_NAME}")
+set(CPACK_DEBIAN_CPP_LIB_DEV_PACKAGE_SUGGESTS "cmake, pkg-config, pkg-conf")
```

Comment:
Should this be using `"${CPACK_DEBIAN_PACKAGE_SUGGESTS}" ` or `"${CPACK_DEBIAN_C_LIB_DEV_PACKAGE_SUGGESTS}"` too?

Reply:
Yes, fixed, thanks.